## Multi layers

A partir do notebook baseline, utilizando-se a mesma sequencia, porém com um modelo multilayer, com mais nós, mais épocas e maior bath_size

A tentativa aqui é força bruta.

In [1]:
import pandas
from keras.models import Sequential
from keras.layers import Dense
from keras.wrappers.scikit_learn import KerasClassifier
from keras.utils import np_utils
from sklearn.model_selection import cross_val_score
from sklearn.model_selection import KFold
from sklearn.preprocessing import LabelEncoder
from sklearn.pipeline import Pipeline

Using TensorFlow backend.


### Leitura e conversão dos dados

In [3]:
# load dataset
dataframe = pandas.read_csv("data/train_v2.csv", header=1)
dataframe = dataframe.apply(pandas.to_numeric, errors='coerce')
dataset = dataframe.values
X = dataset[:,1:770].astype(float)
Y = dataset[:,770]

/Users/jcmlima/opt/anaconda3/envs/bernardo/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3072: DtypeWarning: Columns (135,204,274,417) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [4]:
# encode class values as integers
encoder = LabelEncoder()
encoder.fit(Y)
encoded_Y = encoder.transform(Y)
# convert integers to dummy variables (i.e. one hot encoded)
dummy_y = np_utils.to_categorical(encoded_Y)

dummy_y.shape

(105470, 89)

In [10]:
# define baseline model
def baseline_model():
	model = Sequential()
	model.add(Dense(300, input_dim=769, activation='relu'))
	model.add(Dense(300, input_dim=769, activation='relu'))
	model.add(Dense(300, input_dim=769, activation='relu'))
	model.add(Dense(300, input_dim=769, activation='relu'))
	model.add(Dense(89, activation='softmax'))
	model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])
	return model

In [11]:
estimator = KerasClassifier(build_fn=baseline_model, epochs=15, batch_size=1000, verbose=1)
kfold = KFold(n_splits=5, shuffle=True)

In [12]:
results = cross_val_score(estimator, X, dummy_y, cv=kfold)

Epoch 1/15
84376/84376 [==============================] - 9s 103us/step - loss: nan - accuracy: 0.9056
Epoch 2/15
84376/84376 [==============================] - 7s 87us/step - loss: nan - accuracy: 0.9074
Epoch 3/15
84376/84376 [==============================] - 7s 88us/step - loss: nan - accuracy: 0.9074
Epoch 4/15
84376/84376 [==============================] - 7s 87us/step - loss: nan - accuracy: 0.9074
Epoch 5/15
84376/84376 [==============================] - 7s 88us/step - loss: nan - accuracy: 0.9074
Epoch 6/15
84376/84376 [==============================] - 7s 86us/step - loss: nan - accuracy: 0.9074
Epoch 7/15
84376/84376 [==============================] - 7s 88us/step - loss: nan - accuracy: 0.9074
Epoch 8/15
84376/84376 [==============================] - 7s 87us/step - loss: nan - accuracy: 0.9074
Epoch 9/15
84376/84376 [==============================] - 7s 87us/step - loss: nan - accuracy: 0.9074
Epoch 10/15
84376/84376 [==============================] - 7s 88us/step - loss: n

Finalmente podemos obter a média da acuracidade das 5 execusões, e o desvio padrão entre as 5.

In [13]:
print('-'*100)
print("Baseline: %.2f%% (%.2f%%)" % (results.mean()*100, results.std()*100))

----------------------------------------------------------------------------------------------------
Baseline: 90.72% (0.14%)


Este modelo mais complexo nos permitiu ter a acuracidade de 90.72. Ou seja, a mesma acuracidade do baseline